InfrastructureManager/appscale/infrastructure/__init__.py

```diff
@@ -0,0 +1,522 @@
+
+    """
+    if 'AppScale-Secret' not in self.request.headers \
+        or self.request.headers['AppScale-Secret'] != options.secret:
+      raise CustomHTTPError(HTTPCodes.UNAUTHORIZED, message='Invalid secret')
+
+    parameters = json_decode(self.request.body)
+    logger.info('Operation id received: {}'.format(parameters))
+    if not parameters or 'operation_id' not in parameters:
+      raise CustomHTTPError(HTTPCodes.BAD_REQUEST, message='operation_id is a'
+                                                           'required parameter')
+
+    operation_id = parameters['operation_id']
+    if operation_ids.has_key(operation_id):
+      self.write(json_encode(operation_ids.get(operation_id)))
+    else:
+      logger.error('Operation id not found')
+      raise CustomHTTPError(HTTPCodes.NOT_FOUND, message='Operation id not '
+                                                         'found')
+  @gen.coroutine
+  def post(self):
+    """
+    Spawn new instances using the provided parameters. The request must also
+    contain a 'num_vms' parameter which indicates the number of virtual
+    machines that should be spawned.
+
+    Args:
+      AppScale-Secret: Required in header. Authentication to deployment.
+      args: The request body, a dictionary of values needed to start
+        instances in the specified infrastructure. Infrastructure specific
+        checks are made in  'get_agent', but the following parameters will be
+        verified here:
+          num_vms: Required in body. Number of VMs to start.
+          infrastructure: Required in body, infrastructure to construct an agent
+            for.
+    Returns:
+      If the secret is valid and all the required parameters are available in
+      the input parameter map, this method will return a dictionary containing
+      an 'operation_id' key.
+
+    Raises:
+      CustomHTTPError if the necessary parameters are not filled.
+    """
+    if 'AppScale-Secret' not in self.request.headers \
+        or self.request.headers['AppScale-Secret'] != options.secret:
+      raise CustomHTTPError(HTTPCodes.UNAUTHORIZED, message='Invalid secret')
+
+    args = json_decode(self.request.body)
+
+    logger.info('Received a request to run instances.')
+    for arg in [PARAM_INFRASTRUCTURE, PARAM_NUM_VMS]:
+      if arg not in args:
+        raise CustomHTTPError(HTTPCodes.BAD_REQUEST,
+              message='{} is a required parameter'.format(arg))
+
+    num_vms = int(args[PARAM_NUM_VMS])
+    if num_vms <= 0:
+      logger.warn('Invalid VM count: {0}'.format(num_vms))
+      raise CustomHTTPError(HTTPCodes.BAD_REQUEST,
+                            message='Invalid VM count: {0}'.format(num_vms))
+
+    try:
+      agent, run_params = get_agent(self.agent_factory,
+                                    BaseAgent.OPERATION_RUN, args)
+    except AgentConfigurationException as exception:
+      logger.exception("Error creating agent!")
```

Comment:
Could log some details from the error

Reply:
I think that logger.exception logs the full exception, is that what you mean?